src/command_modules/azure-cli-keyvault/azure/cli/command_modules/keyvault/commands.py

```diff
@@ -41,6 +41,10 @@
                            lambda: keyvault_client_factory().vaults,
                            custom_function_op=custom_path.format('update_keyvault'))
 
+cli_command(__name__, 'keyvault key parse-id', custom_path.format('parse_key_id'))
+cli_command(__name__, 'keyvault secret parse-id', custom_path.format('parse_secret_id'))
+cli_command(__name__, 'keyvault certificate parse-id', custom_path.format('parse_certificate_id'))
```

Comment:
There are many commands which result in entities which only supply id and not name, vault, etc.  Modifying all these commands would be unfeasible.  The ultimate plan is to return these fields from the server so that no customization is needed client side.  Until that happens these provide a workaround which allow CLI users to programatically parse IDs returned from the service without having to either use the SDK directly or take a hard dependency on the format of the IDs.  The commands produce a KeyVaultId object which has the parsed fields vault, name, and version.  As for usage the _help.py and _params.py have been update with proper usage documentation.